### PR TITLE
cleanup waypoint listener

### PR DESCRIPTION
**Pull request by ramaraochavali**
**Please provide a description of this PR:**

### CODE REVIEW

**Review by bleggett — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/listener_waypoint.go [hunk 1]
```diff
@@ -448,29 +448,19 @@ func (lb *ListenerBuilder) waypointInboundRoute(virtualService config.Config, li
...
-					// As an optimization, we can just top sending any more routes here.
-					//if isCatchAllMatch(match) {
-					//	catchall = true
-					//	break
-					//}
-				}
-			}
-		}
-		if catchall {
+			// This is a catchall route, so we can stop processing the rest of the routes.
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener_waypoint.go:456]: looks similar to sidecar

File: pilot/pkg/networking/core/v1alpha3/listener_waypoint.go [hunk 2]
```diff
@@ -448,29 +448,19 @@ func (lb *ListenerBuilder) waypointInboundRoute(virtualService config.Config, li
...
 			break
 		}
+		for _, match := range http.Match {
+			if r := lb.translateRoute(virtualService, http, match, listenPort); r != nil {
+				out = append(out, r)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener_waypoint.go:461]: should we check catch all , if it it catch all, stop the rest.  To keep consistent with sidecar

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/listener_waypoint.go:461]: Not sure. It was like that from the beginning - do not know if it is intentional or a miss. @howardjohn @kyessenov Do you know?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener_waypoint.go:461]: This is alpha implement at that time

**Review by hzxuzhonghu — APPROVED** [review]
LG now

**Pull request merged**